Source/WebCore/dom/TrustedTypeUtils.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2024 Igalia S.L. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "TrustedTypeUtils.h"
+
+#include "ContentSecurityPolicy.h"
+#include "Document.h"
+#include "JSDOMExceptionHandling.h"
+#include "TrustedTypePolicy.h"
+#include "TrustedTypePolicyFactory.h"
+#include "WindowOrWorkerGlobalScopeTrustedTypes.h"
+#include "WorkerGlobalScope.h"
+#include <JavaScriptCore/HeapInlines.h>
+#include <JavaScriptCore/JSCInlines.h>
+#include <JavaScriptCore/JSCJSValueInlines.h>
+#include <JavaScriptCore/JSCast.h>
+
+namespace WebCore {
+using namespace JSC;
+
+struct TrustedTypeVisitor {
+    String operator()(std::nullptr_t)
+    {
+        return nullString();
+    }
+    String operator()(Exception)
+    {
+        return nullString();
+    }
+    String operator()(Ref<TrustedHTML> value)
+    {
+        return value->toString();
+    }
+    String operator()(Ref<TrustedScript> value)
+    {
+        return value->toString();
+    }
+    String operator()(Ref<TrustedScriptURL> value)
+    {
+        return value->toString();
+    }
+};
+
+String trustedTypeToString(const TrustedType trustedType)
```

Comment:
Return type should be `ASCIILiteral` instead of `String`, in some cases this will save creating and then destroying a temporary string.

Argument should be `TrustedType`, not `const TrustedType`.